Sources/CartonCLI/Commands/Test.swift

```diff
@@ -69,49 +72,44 @@ struct Test: AsyncParsableCommand {
   @OptionGroup()
   var buildOptions: BuildOptions
 
-  func buildFlavor() -> BuildFlavor {
-    BuildFlavor(
-      isRelease: release, environment: environment.destination,
-      sanitize: sanitize, swiftCompilerFlags: buildOptions.swiftCompilerFlags
-    )
-  }
-
   func run() async throws {
     let terminal = InteractiveWriter.stdout
-
-    try Self.entrypoint.check(on: localFileSystem, terminal)
     let toolchain = try await Toolchain(localFileSystem, terminal)
-    let flavor = buildFlavor()
-    let testBundlePath = try await toolchain.buildTestBundle(flavor: flavor)
-
-    if environment == .wasmer {
-      terminal.write("\nRunning the test bundle with wasmer:\n", inColor: .yellow)
-      var wasmerArguments = ["wasmer", testBundlePath.pathString]
-      if list {
-        wasmerArguments.append(contentsOf: ["--", "-l"])
-      } else if !testCases.isEmpty {
-        wasmerArguments.append("--")
-        wasmerArguments.append(contentsOf: testCases)
-      }
-
-      try await Process.run(wasmerArguments, parser: TestsParser(), terminal)
-    } else {
-      try await Server(
-        .init(
-          builder: nil,
-          mainWasmPath: testBundlePath,
-          verbose: true,
-          shouldSkipAutoOpen: false,
-          port: port,
-          host: host,
-          customIndexContent: nil,
-          // swiftlint:disable:next force_try
-          manifest: try! toolchain.manifest.get(),
-          product: nil,
-          entrypoint: Self.entrypoint,
-          terminal: terminal
-        )
+    let bundlePath = try await toolchain.buildTestBundle(flavor: buildFlavor())
+
+    switch environment {
+    case .wasmer:
+      try await WasmerTestRunner(
+        testFilePath: bundlePath,
+        listTestCases: list,
+        testCases: testCases,
+        terminal: terminal
+      ).run()
+    case .defaultBrowser:
+      try await BrowserTestRunner(
+        testFilePath: bundlePath,
+        host: host,
+        port: port,
+        // swiftlint:disable:next force_try
+        manifest: try! toolchain.manifest.get(),
+        terminal: terminal
+      ).run()
+    case .node:
+      try await NodeTestRunner(
+        testFilePath: bundlePath,
+        listTestCases: list,
+        testCases: testCases,
+        terminal: terminal
       ).run()
     }
   }
+
+  private func buildFlavor() -> BuildFlavor {
+    BuildFlavor(
+      isRelease: release,
+      environment: environment.destination,
+      sanitize: sanitize,
+      swiftCompilerFlags: buildOptions.swiftCompilerFlags
+    )
+  }
```

Comment:
coding style nit: I realize this was a function before, but let's make this a property, not a function, while we're at it.
```suggestion
  private var buildFlavor: BuildFlavor {
    BuildFlavor(
      isRelease: release,
      environment: environment.destination,
      sanitize: sanitize,
      swiftCompilerFlags: buildOptions.swiftCompilerFlags
    )
  }
```